.travis.yml

```diff
@@ -18,16 +19,22 @@ gemfile:
   - gemfiles/rails_4_2.gemfile
 matrix:
   exclude:
-    - rvm: 2.6.5
+    - rvm: 2.7.1
       gemfile: gemfiles/rails_4_2.gemfile
-    - rvm: 2.4.9
+    - rvm: 2.6.6
+      gemfile: gemfiles/rails_4_2.gemfile
+    - rvm: 2.4.10
       gemfile: gemfiles/rails_6_0.gemfile
-cache: bundler
+cache:
+  directories:
+    - vendor/bundle
 # Source: <https://docs.travis-ci.com/user/languages/ruby/#bundler-20>
 before_install:
-  - gem update --system '2.7.8' --no-document
+  - rm -f ~/.rvm/rubies/ruby-*/bin/bundle
+  - gem update --system --force --no-document
   - gem uninstall -v '>= 2' -i $(rvm gemdir)@global -ax bundler || true
   - gem install bundler -v '< 2' --no-document
+  - bundle config set path vendor/bundle
```

Comment:
Travis sets this by default via the `--path` flag, however, Bundler doesn't seem to remember that setting. This fails integration tests. Integration tests are a bit funky. Each test will create its own Rails application. This application in turn creates its own Gemfile, and so we have to run `bundle install` to set up that application, but we don't actually re-install all of the gems all over again, we assume that all of the gems are already defined in the parent Appraisal gemfile (e.g. `gemfiles/rails_6_0.gemfile`). So effectively whatever gems the Appraisal installs needs to be shared across the tests AND any Rails applications those tests create. By setting the Bundler path in this way we effectively create a global installation path.

Reply:
I lied. This wasn't doing anything :)